sql/src/main/java/io/crate/operation/collect/sources/NodeStatsCollectSource.java

```diff
@@ -63,12 +78,39 @@ public NodeStatsCollectSource(TransportActionProvider transportActionProvider,
         if (collectPhase.whereClause().noMatch()) {
             return ImmutableList.<CrateCollector>of(RowsCollector.empty(downstream));
         }
-
+        Iterator<DiscoveryNode> nodes = clusterService.state().getNodes().iterator();
+        if (collectPhase.whereClause().hasQuery()) {
+            final List<List<Symbol>> exactMatches = this.eqExtractor.extractParentMatches(
+                ImmutableList.<ColumnIdent>builder()
+                    .add(SysNodesTableInfo.Columns.ID)
+                    .build(),
+                collectPhase.whereClause().query(),
+                new StmtCtx()
+            );
+            if (exactMatches != null) {
+                nodes = Iterators.filter(
+                    clusterService.state().getNodes().iterator(),
+                    new Predicate<DiscoveryNode>() {
+                        @Override
+                        public boolean apply(DiscoveryNode input) {
+                            for (List<Symbol> exactMatch : exactMatches) {
+                                for (Symbol match : exactMatch) {
+                                    if (match instanceof Literal && match.valueType().equals(DataTypes.STRING) &&
```

Comment:
Part of this logic could be done outside of the loop.
Maybe it would be possible to re-use part of the logic in `NodeFilters` ?